src/dtx/dtx_rpc.c

```diff
@@ -251,7 +255,8 @@ dtx_req_send(struct dtx_req_rec *drr, daos_epoch_t epoch)
 		drr->drr_tag, req, dra->dra_future,
 		din != NULL ? din->di_epoch : 0, rc);
 
-	if (rc != 0) {
+	if (rc != 0 && drr->drr_comp == 0) {
```

Comment:
I think crt_rpc_send() should guarantee the callback is not called if it returns non-zero, do we need the extra check at all?

Reply:
From current crt_req_send() implementation, if hit failure and returned before setting crp_completee_cb, then we may miss to trigger the callback.

Reply:
sorry I'm confused, I think the definition of crt_seq_send() is: callback is called if returned zero, otherwise it's not called, if it is not correct then it's a bug in cart.

Reply:
Here, the return value is non-zero.